server/src/controllers/orderController.ts

```diff
@@ -143,46 +144,95 @@ orderController.use(checkScopes);
  *       500:
  *         description: Internal Server Error. An error occurred while processing the request.
  */
-orderController.get('/', async (req: Request, res: Response) => {
+orderController.get('/refund', async (req: Request, res: Response) => {
   try {
-    const filters: any = {};
-    if (req.params.ordername) {
-      filters.ordername = {
-        contains: req.params.ordername,
-      };
+    const {email} = req.query;
+    if (!email || typeof email !== 'string') {
+      return res.status(400).send('Email Required');
     }
-    if (req.params.auth0_id) {
-      filters.auth0_id = {
-        contains: req.params.auth0_id,
+    const orders = await prisma.orders.findMany({
+      where: {
+        payment_intent: {not: null},
+        refund_intent: null,
+        ordertotal: {not: 0},
+        contacts: {
+          email: {contains: email},
+        },
+      },
+      select: {
+        orderid: true,
+        ordertotal: true,
+        contacts: {
+          select: {
+            firstname: true,
+            lastname: true,
+            email: true,
+          },
+        },
+        orderdate: true,
+        ordertime: true,
+        orderitems: {
+          select: {
+            singletickets: {
+              select: {
+                ticketwasswapped: true,
+                eventtickets: {
+                  select: {
+                    eventinstances: {
+                      select: {
+                        events: {
+                          select: {
+                            eventname: true,
+                          },
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    });
+    const toReturn = orders.map((order) => {
+      const {
+        contacts,
+        orderitems,
+        ordertotal,
+        orderdate,
+        ...evenythingElse
```

Comment:
This looks like a typo

Reply:
Also, probably want to use a different name

Reply:
fixed